props/blaster.h

```diff
@@ -112,38 +113,43 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     return random <= percent ? true : false;
   }
 
+  virtual void CheckEmpty() {
+    if (max_shots_ != -1 && shots_fired_ >= max_shots_) {
+      empty_ = true;
+      SaberBase::DoEffect(EFFECT_EMPTY, 0);
+      return;
+    }
+  }
+
+  uint32_t auto_time_;
   virtual void Fire() {
 #ifdef ENABLE_MOTION
 #ifdef BLASTER_JAM_PERCENTAGE
     // If we're already jammed then we don't need to recheck. If we're not jammed then check if we just jammed.
     is_jammed_ = is_jammed_ ? true : CheckJam(BLASTER_JAM_PERCENTAGE);
-
-    if (is_jammed_) {
+    if (is_jammed_ && !empty_) {
```

Comment:
I don't think we whould set is_jammed_ either if it's empty.
In fact, maybe the the jam check should be after the empty check below...

Reply:
yes.